message SyncPoint
{
  required int32  robotID        =  1;
  required int32  pointID        =  2;

}

message SyncData
{
  repeated SyncPoint  point  =  1;
}

message Waypoint
{
  enum WayType {
    POS  = 0;
    TIME = 1;
  }
  
  enum VelProfile {
    AROOM   = 0;
    MAMOOLI = 1;
    KHARAKI = 2;
  }
  
  required WayType    type       =  1[default = POS];
  required float      x          =  2;
  required float      y          =  3;
  required bool       needRRT    =  4[default = true];
  required float      speed      =  5[default = 100.0];
  required VelProfile profile    =  6[default = MAMOOLI];
  optional float      tolerance  =  7[default = 100.0];
  optional float      time       =  8;
}

message Role
{
  required int32      id         =  1;
  optional string     name       =  2[default = "NAN"];
  repeated Waypoint   path       =  3;
  required int32      afterlife  =  4;
}

message Strategy
{
  required string     name      =  1;
  repeated Role       role      =  2;
  repeated SyncData   sync      =  3;
  required float	  minX      =  4;
  required float      maxX      =  5;
  required float	  minY      =  6;
  required float      maxY      =  7;

}

message PlayBook
{
  repeated Strategy   strategy  = 1;
  repeated float      weight    = 2;

}